var/spack/repos/builtin/packages/graphviz/package.py

```diff
@@ -93,31 +103,33 @@ def configure_args(self):
         spec = self.spec
         options = []
 
-        # These language bindings have been tested, we know they work.
-        tested_bindings = ('+java', '+perl')
+        need_swig = False
 
-        # These language bindings have not yet been tested.  They
-        # likely need additional dependencies to get working.
-        untested_bindings = (
-            '+swig', '+sharp', '+go', '+guile', '+io',
-            '+lua', '+ocaml', '+php',
-            '+python', '+r', '+ruby', '+tcl')
-
-        for var in untested_bindings:
+        for var in self.untested_bindings:
             if var in spec:
-                raise SpackException(
+                raise InstallError(
                     "The variant {0} for language bindings has not been "
                     "tested.  It might or might not work.  To try it "
                     "out, run `spack edit graphviz`, and then move '{0}' "
                     "from the `untested_bindings` list to the "
                     "`tested_bindings` list.  Be prepared to add "
                     "required dependencies.  "
                     "Please then submit a pull request to "
-                    "http://github.com/llnl/spack")
+                    "http://github.com/llnl/spack".format(var))
+            options.append('--disable-%s' % var[1:])
+
+        for var in self.tested_bindings:
+            if var in spec:
+                need_swig = True
+                enable = 'yes'
+            else:
+                enable = 'no'
+            options.append('--enable-%s=%s' % (var[1:], enable))
```

Comment:
Actually, now that I think about it, why not just do:
```python
for var in self.tested_bindings:
    if var in spec:
        need_swig = True
        options.append('--enable-{0}'.format(var[1:]))
    else:
        options.append('--disable-{0}'.format(var[1:]))
```
Then you don't need to store a variable at all. I feel like that's more inline with how other packages handle it. Although both are valid Python syntax. Sometimes I wish there really was ["one -- and preferably only one -- obvious way to do it"](https://www.python.org/dev/peps/pep-0020/). But you probably find that blasphemous as a Perl programmer 😆